app/models/host/managed.rb

```diff
@@ -537,11 +537,29 @@ def jumpstart?
     operatingsystem.family == "Solaris" and architecture.name =~/Sparc/i rescue false
   end
 
+  HOSTGROUP_INHERITED_ATTRIBUTES = %w{puppet_proxy_id puppet_ca_proxy_id environment_id compute_profile_id}
+
+  def apply_inherited_attributes(attributes)
+    new_hostgroup_id = attributes['hostgroup_id']
+    return attributes if new_hostgroup_id.nil?
```

Comment:
so the question here was for existing host, that is removed from hostgroup whether it make sense to nilify the attributes that were inherited (we could match the actual value) but since they were copied and the change on hostgroup does not affect the host value already I was not sure whether it make sense, probably not